source/game/kart/KartMove.cc

```diff
@@ -1384,6 +1402,166 @@ void KartMove::landTrick() {
     activateBoost(KartBoost::Type::TrickAndZipper, duration);
 }
 
+/// @addr{0x8058498C}
+void KartMove::enterCannon() {
+    init(true, false);
+    physics()->clearDecayingRot();
+    m_boost.resetActive();
+    state()->setBoost(false);
+
+    m_jumpPadMinSpeed = 0;
+    m_rampBoost = 0;
+    state()->setRampBoost(false);
+    m_ssmtCharge = 0;
+    m_ssmtLeewayTimer = 0;
+    m_ssmtDisableAccelTimer = 0;
+    m_offroadInvincibility = 0;
+
+    state()->setBoostOffroadInvincibility(false);
+
+    dynamics()->reset();
+
+    state()->setHop(false);
+    state()->setInCannon(true);
+    state()->setSkipWheelCalc(true);
+    state()->setCannonStart(false);
+
+    const auto [cannonPos, cannonDir] = cannonPosRot();
+    const EGG::Vector3f kartPos = this->pos();
```

Comment:
```suggestion
     m_cannonEntryPos = m_pos;
    m_cannonEntryOfs = cannonPos - m_pos;
```

Reply:
`m_pos` is a member of dynamics not move, so i did this:
```cpp
    m_cannonEntryPos = pos();
    m_cannonEntryOfs = cannonPos - pos();
```

Reply:
Yes you're right, sorry I was very tired last night LOL